### PR TITLE
Add a link to catalog bento results to the Database format facet…

### LINKED ISSUE #82: Add the "Your search also found..." text/link in catalog bento

### PR DESCRIPTION
**Pull request by jkeck**
… (with hits).

Closes #82 

climate-change

### CODE REVIEW

File: app/services/catalog_search_service.rb [hunk 1]
```diff
@@ -30,6 +31,14 @@ def facets
       json['response']['facets']
     end
 
+    def additional_facet_details(q)
+      return nil if additional_facet_item['hits'].to_i.zero?
+      Struct.new(:hits, :href).new(
+        additional_facet_item['hits'],
+        "#{QUERY_URL % { q: q }}&f[#{HIGHLIGHTED_FACET_FIELD}][]=#{ADDITIONAL_FACET_TARGET}"
```
Comment (drh-stanford) [inline app/services/catalog_search_service.rb]: i think we'll need `q: CGI.escape(q)` here.

File: spec/services/catalog_search_service_spec.rb [hunk 2]
```diff
@@ -12,5 +12,40 @@
...
+          expect(addl_facet_details.hits).to eq 123
+        end
+
+        it 'has a href that links to the full query' do
+          expect(addl_facet_details.href).to include('/?q=blah&f[format_main_ssim][]=Database')
+        end
+      end
+
+      context 'when the facet does not have any hits' do
+        let(:facet_data) { [{ 'name' => 'format_main_ssim', 'items' => [{ 'label' => 'Database', 'hits' => '0' }]}] }
```
Comment (drh-stanford) [inline spec/services/catalog_search_service_spec.rb]: should this be `'hits' => 0` to match the Solr response?

File: app/views/quick_search/search/_module.html.erb [hunk 3]
```diff
@@ -23,6 +23,14 @@
         <ol>
             <%= render partial: '/quick_search/search/result', collection: searcher.results %>
         </ol>
+
+        <% if (addl_facet = searcher.search.additional_facet_details(params[:q])) %>
+          <p class='text-center'>
+            Your search also found
+            <%= link_to("#{addl_facet.hits} topic specific databases", addl_facet.href) %>.
```
Comment (drh-stanford) [inline app/views/quick_search/search/_module.html.erb]: do we need to format this number (hits) with commas, etc.?

File: spec/services/catalog_search_service_spec.rb (hunk 2 above)
Comment (jkeck) [reply spec/services/catalog_search_service_spec.rb]: I guess this was kind of a hidden test for string numbers are cast to integers.  Happy to change this or make that test more explicit.

Comment (drh-stanford) [reply spec/services/catalog_search_service_spec.rb]: ok that's fine

**Comment by jkeck** [comment]
@drh-stanford I have updated and addressed the conflicts as well.  Please let me know if you see anything else that needs fixing.

**Pull request merged**